src/Core/Checkout/Document/Zugferd/ZugferdDocument.php

```diff
@@ -204,13 +204,14 @@ public function withDiscountItem(OrderLineItemEntity $lineItem): self
         $isPercentage = (($lineItem->getPayload()['discountType'] ?? null) === PromotionDiscountEntity::TYPE_PERCENTAGE)
             && (abs($lineItem->getTotalPrice()) !== (float) ($lineItem->getPayload()['maxValue'] ?? null));
 
-        $this->addMappedPrice(self::ALLOWANCE_AMOUNT, $lineItem->getPrice());
+        $type = $lineItem->getPrice()->getUnitPrice() >= 0 ? self::CHARGE_AMOUNT : self::ALLOWANCE_AMOUNT;
+        $this->addMappedPrice($type, $lineItem->getPrice());
 
         foreach ($lineItem->getPrice()->getCalculatedTaxes() as $calculatedTax) {
             $actualAmount = $this->getPrice($calculatedTax);
 
             if (!Feature::isActive('v6.8.0.0')) {
-                $this->addAllowanceAmount($actualAmount);
+                $actualAmount >= 0 ? $this->addChargeAmount($actualAmount) : $this->addAllowanceAmount($actualAmount);
```

Comment:
Just a small nitpick: I would prefer an real if statement here because you have no assignment of the return values and calling different code based on a condition